util/util.go

```diff
@@ -1,3 +1,42 @@
 package util
 
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+
+	"github.com/cloudfoundry-community/go-cfclient"
+	"github.com/hashicorp/go-cleanhttp"
+	"github.com/hashicorp/vault-plugin-auth-pcf/models"
+)
+
 const BashTimeFormat = "Mon Jan 2 15:04:05 MST 2006"
+
+// NewPCFClient does some work that's needed every time we use the PCF client,
+// namely using cleanhttp and configuring it to match the user conf.
+func NewPCFClient(config *models.Configuration) (*cfclient.Client, error) {
```

Comment:
This was added to encapsulate properly making a PCF client, which is done multiple times.